checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakChecker.java

```diff
@@ -93,4 +175,103 @@ public void typeProcessingOver() {
     }
     super.typeProcessingOver();
   }
+
+  /**
+   * Get the set of exceptions that should be ignored. This set comes from the {@link
+   * #IGNORED_EXCEPTIONS} option if it was provided, or {@link #DEFAULT_IGNORED_EXCEPTIONS} if not.
+   *
+   * @return the set of exceptions to ignore
+   */
+  public SetOfTypes getIgnoredExceptions() {
+    SetOfTypes result = ignoredExceptions;
+    if (result == null) {
+      String ignoredExceptionsOptionValue = getOption(IGNORED_EXCEPTIONS);
+      result =
+          ignoredExceptionsOptionValue == null
+              ? DEFAULT_IGNORED_EXCEPTIONS
+              : parseIgnoredExceptions(ignoredExceptionsOptionValue);
+      ignoredExceptions = result;
+    }
+    return result;
+  }
+
+  /**
+   * Parse the argument given for the {@link #IGNORED_EXCEPTIONS} option. Warnings will be issued
+   * for any problems in the argument, for instance if any of the named exceptions cannot be found.
+   *
+   * @param ignoredExceptionsOptionValue the value given for {@link #IGNORED_EXCEPTIONS}
+   * @return the set of ignored exceptions
+   */
+  protected SetOfTypes parseIgnoredExceptions(String ignoredExceptionsOptionValue) {
+    String[] exceptions = COMMAS.split(ignoredExceptionsOptionValue);
+    List<SetOfTypes> sets = new ArrayList<>();
+    for (String e : exceptions) {
+      SetOfTypes set = parseExceptionSpecifier(e, ignoredExceptionsOptionValue);
+      if (set != null) {
+        sets.add(set);
+      }
+    }
+    return SetOfTypes.union(sets.toArray(new SetOfTypes[0]));
+  }
+
+  /**
+   * Parse a single exception specifier from the {@link #IGNORED_EXCEPTIONS} option and issue
+   * warnings if it does not parse. See {@link #EXCEPTION_SPECIFIER} for a description of the
+   * syntax.
+   *
+   * @param exceptionSpecifier the exception specifier to parse
+   * @param ignoredExceptionsOptionValue the whole value of the {@link #IGNORED_EXCEPTIONS} option;
+   *     only used for error reporting
+   * @return the parsed set of types, or null if the value does not parse
+   */
+  protected @Nullable SetOfTypes parseExceptionSpecifier(
+      String exceptionSpecifier, String ignoredExceptionsOptionValue) {
+    Matcher m = EXCEPTION_SPECIFIER.matcher(exceptionSpecifier);
+    if (m.matches()) {
+      @Nullable String questionMarkExtendsClause = m.group(1);
+      String qualifiedName = m.group(2);
+
+      if (qualifiedName.equalsIgnoreCase("default")) {
+        return DEFAULT_IGNORED_EXCEPTIONS;
+      }
+      TypeMirror type = checkCanonicalName(qualifiedName);
+      if (type == null) {
```

Comment:
It seems hypothetically possible that a user would want to ignore exceptions of a type that is not in the classpath (e.g., if the only methods that can throw that exception are in 3rd-party jars).  But, the additional validity checking we get here is probably worth not supporting that case.  Maybe we can at least add a comment around this, though?

Reply:
Actually, this is a great point. I will adjust this path so that the name is still added to the set of ignored exceptions, even if it doesn't seem to reference a real type. (But, I will retain the warning because I think it is important to catch typos in the option value.)